scripts/dev/upstream_merge/build_and_install_kernel.py

```diff
@@ -0,0 +1,291 @@
+    kernel_parent_dir = os.path.dirname(config.kernel_src_dir)
+    os.chdir(kernel_parent_dir)
+    try:
+        kernel_version = config.target_branch.split('/')[-1] if config.target_branch else "6.12"
+        git_obj = GitRepo(
+            local_repo=os.path.basename(config.kernel_src_dir),
+            upstream_repo_url=config.stable_rt_remote,
+            upstream_branch=f"linux-{kernel_version}.y-rt",
+            local_base_branch=config.target_branch,
+            upstream_repo_name="stable-rt",
+            fork_name="origin",
+            fork_url=config.repo_url
+        )
+        os.chdir(config.kernel_src_dir)
+        try:
+            git_merge_abort()
+        except:
+            pass
+        git_reset(hard=True)
+        git_clean(force=True, directories=True, ignored_files=True)
+        status, _ = git_fetch()
+        if status != 0:
+            raise RuntimeError("Failed to fetch remotes")
+        status, _ = run_command(f"git checkout -B {config.target_branch} origin/{config.target_branch}")
+        if status != 0:
+            raise RuntimeError("Failed to checkout target branch")
+        status, remotes = git_remote()
+        if status == 0 and "stable-rt" not in remotes:
+            git_obj.add_remote("stable-rt", config.stable_rt_remote)
+        git_fetch("stable-rt", "--tags")
+        status, tags = git_tag(list_pattern=f"v{kernel_version}.*-rt*")
+        if status != 0 or not tags:
+            print(f"[WARNING] No v{kernel_version}-rt tags; skipping merge")
+            return
+        clean_tags = [t for t in tags.splitlines() if re.match(rf'^v{re.escape(kernel_version)}\.\d+(?:\.\d+)?-rt\d+$', t)]
+        if not clean_tags:
+            print(f"[WARNING] No clean v{kernel_version}-rt release tags; skipping merge")
+            return
+        latest_tag = sorted(clean_tags, key=lambda t: list(map(int, re.findall(r'\d+', t))))[-1]
+        print(f"[INFO] Latest RT tag: {latest_tag}")
+        merge_result = git_obj.merge_branch(latest_tag, f"Merge latest upstream {latest_tag}")
+        if merge_result[0] != 0:
+            status, conflicts = git_diff(name_only=True, diff_filter="U")
+            if status == 0 and conflicts:
+                status, git_status_output = git_status()
+                concise_body = _format_conflict_email(latest_tag, conflicts, git_status_output, config.target_branch)
+                send_email_report(
+                    f"Merge conflict report: {config.target_branch}",
+                    concise_body
+                )
+                raise RuntimeError("[ERROR] Merge conflicts detected; email sent")
+        else:
+            print("[INFO] Merge successful")
+    finally:
+        os.chdir(original_cwd)
+
+# === Step 3: Build and Deploy Kernel ===
+def build_and_deploy_kernel(args):
+    try:
+        if not args.skip_packages:
+            check_required_packages()
+        CROSS_COMPILE = detect_or_build_cross_compile(config, script_dir)
```

Comment:
Aren't `config` and `script_dir` already global? Why pass them to another function as parameters?